core/trino-spi/src/main/java/io/trino/spi/function/AggregationFunction.java

```diff
@@ -13,13 +13,16 @@
  */
 package io.trino.spi.function;
 
+import com.google.errorprone.annotations.Keep;
+
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
 import static java.lang.annotation.ElementType.METHOD;
 import static java.lang.annotation.ElementType.TYPE;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
+@Keep
```

Comment:
Annotate `@UsedByGeneratedCode` too.

And perhaps add a commit message explaining why.